src/System.Text.Json/tests/Serialization/CyclicTests.cs

```diff
@@ -19,6 +19,43 @@ public static void WriteCyclicFail()
             Assert.Throws<JsonException>(() => JsonSerializer.ToString(obj));
         }
 
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(10)]
+        [InlineData(70)]
+        public static void WriteCyclicFail(int depth)
+        {
+            var rootObj = new TestClassWithCycle("root");
+            CreateObjectHierarchy(1, depth, rootObj);
+
+            {
+                var options = new JsonSerializerOptions();
+                options.MaxDepth = depth + 1;
```

Comment:
What happens if a user sets this to 1000 or even larger?

Reply:
If larger, the 1,000 limit on the writer will kick in.

Reply:
Hmm... So, we can't serialize any objects > 64 depth, even though we can write JSON with depth > 64? Seems like the limitation is too strict here.